k9mail/src/main/java/com/fsck/k9/activity/setup/AccountSettings.java

```diff
@@ -733,6 +749,28 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
         }
     }
 
+    private void showApgDeprecationDialog() {
+        ApgDeprecationWarningDialog fragment = ApgDeprecationWarningDialog.newInstance();
```

Comment:
I tried to use `showDialog`, and it's even worse - on orientation change the dialog is recreated, but the preference picker is shown on top of it. I think this has to do with the way fragments are restored by the PreferenceActivity, but I'm not sure.

I changed the dismiss listener structure now to solve the activity leak more elegantly, but I don't think investing more hours into this super corner case ux quirk is worth it.